packages/fx-core/src/component/driver/botAadApp/create.ts

```diff
@@ -98,12 +99,25 @@ export class CreateBotAadAppDriver implements StepDriver {
         throw new UnexpectedEmptyBotPasswordError(actionName, helpLink);
       }
 
+      const tokenJson = await context.m365TokenProvider.getJsonObject({ scopes: GraphScopes });
+      const isMsftAccount =
+        tokenJson.isOk() &&
+        tokenJson.value.unique_name &&
+        (tokenJson.value.unique_name as string).includes("@microsoft.com");
+
       const startTime = performance.now();
       if (!botAadAppState.botId) {
         context.logProvider?.info(getLocalizedString(logMessageKeys.startCreateBotAadApp));
+
+        // This hidden environment variable is for internal use only.
+        const serviceManagementReference = process.env.TTK_DEFAULT_SERVICE_MANAGEMENT_REFERENCE;
+        if (isMsftAccount && !serviceManagementReference) {
+          throw new MissingServiceManagementReferenceError(actionName);
```

Comment:
so the reference all from ENV? is there support file?

Reply:
We now recommend using `aadApp/create` over the `botAadApp/create` action. The help doc will ask user to change `botAadApp/create` to `aadApp/create` to provide service management reference. `botAadApp/create` won't be updated to support service management reference. `aadApp/create` is updated a long time ago with the service management reference support.
Since we hope the env can set default service management reference, I only include env related logic here.